### PR TITLE
[v9] Add OS compatibility checker

### PR DESCRIPTION
**Pull request by jakule**
Backport of https://github.com/gravitational/teleport/pull/14539 https://github.com/gravitational/teleport/pull/15547 https://github.com/gravitational/teleport/pull/16032

### CODE REVIEW

File: .cloudbuild/ci/os-compatibility-test.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+timeout: 25m
+
+options:
+   machineType: E2_HIGHCPU_32
+
+steps:
+  - name: public.ecr.aws/gravitational/teleport-buildbox-centos7:teleport11
```
Comment (greedy52) [inline .cloudbuild/ci/os-compatibility-test.yaml:7]: just realize we are using `teleport11` . should we use `teleport9`? Totally missed in other backport

Comment (jakule) [reply .cloudbuild/ci/os-compatibility-test.yaml]: Good point. I've fixed the tag in this image. I'll create a matching PR in v10 in a sec.

Comment (jakule) [reply .cloudbuild/ci/os-compatibility-test.yaml]: v10 PR https://github.com/gravitational/teleport/pull/16177

**Review by r0mant — APPROVED** [review]

**Review by greedy52 — APPROVED** [review]

**Comment by jakule** [comment]
Closed in favor of https://github.com/gravitational/teleport/pull/16262

**Pull request closed without merging**